### ISSUE TITLE
Unit tests for new_student view

### ISSUE THREAD

**Issue by garg3133** [garg3133/JagratiWebApp#79]
Write unit tests for `new_student` view in the `students` app. The tests need to be added in a file `test_views` in `test` directory of `students` app.

To learn about how to write unit tests in Django, you may follow this tutorial from MDN: https://developer.mozilla.org/en-US/docs/Learn/Server-side/Django/Testing

For any further doubts and queries, feel free to join the community and ask there: https://discord.gg/Ek9q45ZjAv

**Comment by Vrindagupta6828** [comment]
@garg3133 Can I work on this issue. Please assign this issue to me.

**Comment by garg3133** [comment]
Sure @Vrindagupta6828!

**Comment by Vrindagupta6828** [comment]
@garg3133 Can you please tell me how to run the test written as when i am executing the statement manage.py test students it is showing the following error:
ModuleNotFoundError: No module named 'students'

**Comment by garg3133** [comment]
Try to run `python manage.py test apps/students`

**Comment by Vrindagupta6828** [comment]
I did all that but when i am running this it is showing ran 0 test

**Comment by Vrindagupta6828** [comment]
It should show ran 1 test

**Comment by garg3133** [comment]
Have you deleted the `tests.py` file? If not, you should delete it.

**Comment by Vrindagupta6828** [comment]
I have deleted the file and create a folder named tests and in that have made a fike test_views.py

**Comment by Vrindagupta6828** [comment]
@garg3133 Can you please assist me how to advance further?

**Comment by garg3133** [comment]
@Vrindagupta6828 Where are you stuck at?

You first need to create a class with name that matches function view name like `NewStudentViewTest`.

In that class, create a function `setUpTestData` in which you'll need to create some sample user (with `auth` & `is_active` as `True`), and then create that user's volunteer profile so that the user can access `new_student` view.

Then the first few functions will be same as in https://github.com/garg3133/fossee/blob/master/roomslotbooking/home/tests/test_views.py IndexViewTest which just tests if the user is being redirected properly if the user is not a volunteer, hasn't completed the profile or not authenticated. Then it should test if the correct template is being shown to the user. Then just the basic working of the `new_student` view to check if everything is working as expected.

Everything is explained in great details in this tutorial: https://developer.mozilla.org/en-US/docs/Learn/Server-side/Django/Testing

Testing is a very major part of any software and hence should be done properly. 

You can watch this video to get an essence of why we actually do testing: https://www.youtube.com/watch?v=1Lfv5tUGsn8

**Comment by anurag0singh** [comment]
@garg3133 please assign this to me under GSSOC'21

**Comment by garg3133** [comment]
Sure @anurag0singh!

**Comment by harshakhmk** [comment]
Is the issue still alive to add code, I am interested in adding the unit-tests

**Comment by anurag0singh** [comment]
> Is the issue still alive to add code, I am interested in adding the unit-tests

Yes, you can work on it.
@garg3133  I am sorry but I cannot do this work. Please assign @harshakhmk on this one.

**Comment by AJAYSHARMA2012** [comment]
> Is the issue still alive to add code, I am interested in adding the unit-tests

@harshakhmk I am assigning you this issue

**Comment by harshakhmk** [comment]
Do I need to add test views in all apps

**Comment by harshakhmk** [comment]
I am working on it @Jagrati-Bot

**Comment by garg3133** [comment]
> Do I need to add test views in all apps

No, you just need to add the tests in the `students` app for this issue.

**Comment by harshakhmk** [comment]
I have created the PR  [262](https://github.com/garg3133/JagratiWebApp/pull/262) please review it @garg3133

**Comment by harshakhmk** [comment]
@Jagrati-Bot  review pending

**Comment by harshakhmk** [comment]
I am sorry, I was not able to work @priyansh3133  you guided me well with the workflow, but it failed on my local system and I was not clear with flow of user signup process, unassign me this issue

**Comment by akshay399** [comment]
@garg3133  is this issue open? if so pls assign me, i have never worked with unit tests i want to give it a try
